packages/nimble-components/src/select/specs/filterable-select-hld.md

```diff
@@ -20,29 +20,58 @@ Our clients have a need of a filterable dropdown component that does not allow a
 
 #### Select
 
-We will provide a means for clients to enable this feature with a new attribute:
+We will provide a means for clients to enable this feature with some new attributes:
 
 ```ts
 export class Select() {
     ...
     @attr({ attribute: 'filter-mode' })
     public filterMode = FilterMode.none;
+
+    /**
+     * Displays a message at the bottom of the dropdown to indicate more options
+     * are currently being loaded.
+     */
+    @attr({ attribute: 'is-loading' })
+    public isLoading = false;
     ...
 }
 
 // Provide enum for filterMode to allow for future modes
 export const FilterMode  = {
     none: undefined;
     standard: 'standard';
+    custom: 'custom';
```

Comment:
Curious if you considered other words rather than `custom` here? Specifically, I'm wondering if `manual` would be a better term since you are requiring the client to "manually" filter the options. My only [slight] concern with "custom" is that it could be read to mean that a custom filter function/regex will be provided for matching, and that could either lead to client confusion or limit our API choices in the future.

Reply:
I'm not against the term `manual`. @jattasNI, @rajsite, thoughts?

Reply:
I'm struggling to think of existing native or Nimble APIs that we could use as precedence. So I'm left with personal opinions. I like `custom` marginally better than `manual` with the not-very-compelling rationale that "manual" for me has connotations of a human tediously filtering items by hand. 

Some other ideas:
- `filter-mode="event"` - you filter by responding to events. `input-event` or `filter-input-event` are possible variations
- `filter-mode="dynamic"` - the items can change dynamically. `programmatic` is a possible variation
- `filter-mode="client"` - the client provides the filtering. `client-provided` or `user-defined` are possible variations.

I'm not seeing a clear winner. My personal preferences are `dynamic` or `custom`.

Reply:
I like `dynamic`, especially over `custom`. The places we use `custom` today tend to be in table columns where a number of other `custom-<configuration-option>` attributes are also used to refine the "custom" behavior. That's pretty different than what we're doing here.

Reply:
Going with `dynamic`.

Reply:
Sorry to hop on late but to me "dynamic" is in contrast to "static" so as just a name it's not intuitive to me what filter-mode "dynamic" vs "standard" vs "none" mean.

What is dynamic is how the user is expected to treat options, but really options are always "dynamic"; we only show options that are slotted and what we show changes regardless of this attribute configuration.

Here the filtering is really "disabled". We aren't doing dynamic filtering, we are doing no filtering.

Reply:
I'd propose going with either `manual` or `custom` with the options given. And given the above examples of how we use custom to tweak config instead of completely change, I have a slight preference for `manual`

Reply:
I can live with the emerging consensus of `manual`